sallyport/src/lib.rs

```diff
@@ -171,6 +171,41 @@ impl Block {
     pub const fn buf_capacity() -> usize {
         Page::size() - size_of::<Message>()
     }
+
+    #[allow(unused)]
+    fn cursor<'a, E>(&'a mut self, err: E) -> Cursor<'a, E> {
+        Cursor {
+            ptr: &mut self.buf,
+            err,
+        }
+    }
+}
+
+/// Helper for allocation of untrusted memory in a Block.
+pub struct Cursor<'a, E> {
+    ptr: &'a mut [u8],
+    err: E,
+}
+
+impl<'a, E> Cursor<'a, E> {
+    /// Allocates a `count`-sized chunk of a buffer for use. And updates the buffer's pointer.
```

Comment:
This documentation is incorrect. It allocates an array, containing `count` number of T items. The result is uninitialized. This needs to be documented in a `# Safety` section.